packages/core/src/common/keybinding.ts

```diff
@@ -180,11 +194,33 @@ export class KeybindingRegistry {
      * @param keyCode The key code for which we are looking for keybindings.
      */
     getKeybindingsForKeyCode(keyCode: KeyCode): Keybinding[] {
-        const bindings = this.keybindings[keyCode.keystroke] || [];
-
-        this.sortKeybindingsByPriority(bindings);
-
-        return bindings;
+        const result: Keybinding[] = [];
+
+        for (let scope = KeybindingScopes.DEFAULT; scope < KeybindingScopes.END; scope++) {
+            this.keymaps[scope].forEach(binding => {
+                if (KeyCode.equals(binding.keyCode, keyCode)) {
+
+                    const _isMoreSpecificBinding: (genericScope: number, matchingBinding: Keybinding) => boolean = (genericScope: number, matchingBinding: Keybinding) => {
+                        if (genericScope === KeybindingScopes.END) {
+                            return false;
+                        }
+
+                        const specificBindings = this.keymaps[genericScope + 1].filter(specificBinding => matchingBinding.commandId === specificBinding.commandId);
```

Comment:
Ah now "extracting it makes more sense! `getKeybindingsForCommand` would also use it, kinda like how `isShadowed` was before with the index.

Reply:
Also I think you meant  `let nextScope = ++scope;` otherwise it checks for the same scope (instead of checking for the more specific scope). I just pushed a new patch with the changes (and the correct prefix increment).